app/uk/gov/hmrc/soletraderidentificationfrontend/services/SubmissionService.scala

```diff
@@ -25,61 +25,163 @@ import javax.inject.{Inject, Singleton}
+        .retrieveIndividualDetails(journeyId)
+        .map(_.getOrElse(noDataServerException(journeyId)))
+
+      matchingResult <- calculateMatchingResult(journeyId, journeyConfig, individualDetails)
+
+      response <- calculateSubmissionResponse(
+        journeyId,
+        calculateBusinessScenario(journeyConfig),
+        matchingResult,
+        journeyConfig,
+        individualDetails
       )
-      matchingResult <-
-        if (individualDetails.optNino.isEmpty && isEnabled(EnableNoNinoJourney))
-          soleTraderMatchingService.matchSoleTraderDetailsNoNino(journeyId, individualDetails)
-        else soleTraderMatchingService.matchSoleTraderDetails(journeyId, individualDetails, journeyConfig)
-      response <- matchingResult match {
-        case Right(true) if individualDetails.optSautr.nonEmpty =>
-          individualDetails.optSautr match {
-            case Some(sautr) => businessVerificationService.createBusinessVerificationJourney(journeyId, sautr).flatMap {
-              case Right(BusinessVerificationJourneyCreated(businessVerificationUrl)) =>
-                Future.successful(StartBusinessVerification(businessVerificationUrl))
-              case _ =>
-                for {
-                  _ <- soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
-                } yield {
-                  JourneyCompleted(journeyConfig.continueUrl)
-                }
-            }
-          }
-        case Right(_) if !journeyConfig.pageConfig.enableSautrCheck =>
-          Future.successful(JourneyCompleted(journeyConfig.continueUrl))
-        case Right(_) if individualDetails.optNino.isEmpty && isEnabled(EnableNoNinoJourney) => for {
-          _ <- createTrnService.createTrn(journeyId)
-          _ <- soleTraderIdentificationService.storeBusinessVerificationStatus(journeyId, BusinessVerificationUnchallenged)
-          _ <- soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
-        } yield {
-          JourneyCompleted(journeyConfig.continueUrl)
-        }
-        case Right(_) => for {
-          _ <- soleTraderIdentificationService.storeBusinessVerificationStatus(journeyId, BusinessVerificationUnchallenged)
-          _ <- soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
-        } yield {
-          JourneyCompleted(journeyConfig.continueUrl)
-        }
-        case Left(failureReason) =>
+    } yield
+      response
+
+  private def noDataServerException(journeyId: String): Nothing =
+    throw new InternalServerException(s"Details could not be retrieved from the database for $journeyId")
+
+  private def calculateBusinessScenario(journeyConfig: JourneyConfig): BusinessScenario =
+    if (!journeyConfig.pageConfig.enableSautrCheck) IndividualJourney
+    else {
+      if (journeyConfig.businessVerificationCheck) SoleTraderJourneyWithSaUtrCheck
```

Comment:
Good spot, I agree here

Reply:
ok I will change the name then

Reply:
Richard Milly let me know if this is okay otherwise write them and I will use them ;-)

 private sealed trait BusinessScenario

  private case object SoleTraderJourneyWithoutBusinessVerification extends BusinessScenario

  private case object SoleTraderJourneyWithBusinessVerification extends BusinessScenario

  private case object IndividualJourney extends BusinessScenario

Reply:
I liked the term skip .... because it has been used in the jira ticket and it is shorter .... please tell me the names if you dont like the one above